Feat: Clothes Crawling 및 Controller 구현

dmdwn99 opened the pull request:
## 📕 제목
- #35

## 📗 작업 내용

> 구현 내용 및 작업 했던 내역

- [x] 작업내용1: ClothesGroup, ClothesType 엔티티 구현
- [x] 작업내용2: ClothesGroupType 엔티티를 통해 다대다 연관관계 매핑
- [x] 작업내용3: ClothesType 중복 저장을 위해 복합 기본 키 설정
- [x] 작업내용4: Top, Bottom Crawling 및 Save 구현
- [x] 작업내용5: ClothesController 구현

## 📘 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 특이사항1: Crawling Scheduling은 제외 하였음
- 특이사항2: ClothesGroup, ClothesType, ClothesGroupType 엔티티 연관관계 매핑 확인
- 특이사항3: Top, Bottom 추천 알고리즘 미 구현으로 RandomSorting으로 구현
-

dmdwn99 committed:
Merge remote-tracking branch 'origin/develop' into feat/clothes-crawling

dmdwn99 committed:
Merge branch 'develop' into feat/clothes-crawling

dmdwn99 committed:
Merge branch 'develop' into feat/clothes-crawling

dmdwn99 committed:
Merge remote-tracking branch 'origin/develop' into feat/clothes-crawling

dmdwn99 committed:
Merge remote-tracking branch 'origin/develop' into feat/clothes-crawling

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현

1. Clothes 관련 엔티티 Of 메서드 구현
2. ClothesController 구현 및 DTO 구현
3. ClothesName value 수정
4. 성별별 Top, Bottom crawling 구현

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Clothes 관련 엔티티 Of 메서드 구현
2. ClothesController 구현 및 DTO 구현
3. ClothesName value 수정
4. 성별별 Top, Bottom crawling 구현

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Clothes 관련 엔티티 Of 메서드 구현
2. ClothesController 구현 및 DTO 구현
3. ClothesName value 수정
4. 성별별 Top, Bottom crawling 구현

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Clothes 관련 엔티티 Of 메서드 구현
2. ClothesController 구현 및 DTO 구현
3. ClothesName value 수정
4. 성별별 Top, Bottom crawling 구현

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Clothes 관련 엔티티 Of 메서드 구현
2. ClothesController 구현 및 DTO 구현
3. ClothesName value 수정
4. 성별별 Top, Bottom crawling 구현

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Clothes 관련 엔티티 Of 메서드 구현
2. ClothesController 구현 및 DTO 구현
3. ClothesName value 수정
4. 성별별 Top, Bottom crawling 구현

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Clothes 관련 엔티티 Of 메서드 구현
2. ClothesController 구현 및 DTO 구현
3. ClothesName value 수정
4. 성별별 Top, Bottom crawling 구현

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. ClothesGroup, ClothesType 엔티티 구현
2. ClothesGroupType 엔티티를 통해 다대다 연관관계 매핑
3. ClothesType 중복 저장을 위해 복합 기본 키 설정

dmdwn99 committed:
Merge remote-tracking branch 'origin/develop' into feat/clothes-crawling

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Ubuntu Server에 맞게 driver 설정 변경

dmdwn99 committed:
Feat: Clothes Crawling 및 Controller 구현 #35

1. Ubuntu Server에 맞게 driver 설정 변경

reeruryu commented on src/main/java/com/seungah/todayclothes/domain/clothes/dto/response/TopResponse.java:
```diff
@@ -0,0 +1,23 @@
+@NoArgsConstructor
+@AllArgsConstructor
+public class TopResponse {
+    private String itemUrl;
```
id 추가 부탁드려요!

reeruryu replied:
BottomResponse도요~!

reeruryu commented on src/main/java/com/seungah/todayclothes/domain/clothes/service/ClothesService.java:
```diff
@@ -0,0 +1,169 @@
+    public ResponseEntity<List<TopResponse>> getTopClothes(Integer groupNumber) {
+        List<TopResponse> topResponseList = new ArrayList<>();
+
+        List<Top> topList = topRepository.findByClothesGroup(clothesGroupRepository.findByGroupNumber(groupNumber));
```
findByGroupNumber null 처리하면 좋을 것 같아용

reeruryu commented on src/main/java/com/seungah/todayclothes/domain/clothes/service/ClothesService.java:
```diff
@@ -0,0 +1,169 @@
+        return new ChromeDriver(options);
+    }
+
+    public ResponseEntity<List<TopResponse>> getTopClothes(Integer groupNumber) {
```
Crawling이랑 clothesgroup으로 조회하는 메서드는 나누면 더 보기 좋을 것 같습니다!

reeruryu commented on src/main/java/com/seungah/todayclothes/domain/clothes/service/ClothesService.java:
```diff
@@ -0,0 +1,169 @@
+    public void saveManClothes(String type, String itemUrl, String imageUrl){
+        boolean isDuplicate = topRepository.existsByItemUrl(itemUrl) || bottomRepository.existsByItemUrl(itemUrl);
+        if (!isDuplicate) {
+            ClothesType clothesType = clothesTypeRepository.findByGenderAndType(Gender.MALE, type);
```
null 처리하면 좋을 것 같아용

reeruryu reviewed (commented):
👍 👍 👍 👍 👍 👍 👍 👍 고생많았다..

dmdwn99 committed:
Refactor: Clothes Crawling 및 Controller 구현 #35

1. TopResponse, BottomResponse id 추가
2. Crawling, api 요청 Service 분리
3. findbyClothesGroup 예외처리
